src/Symfony/Component/Intl/Data/Generator/TimezoneDataGenerator.php

```diff
@@ -0,0 +1,163 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function preGenerate()
+    {
+        $this->timezones = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForLocale(BundleReaderInterface $reader, $tempDir, $displayLocale)
+    {
+        $localeBundle = $reader->read($tempDir, $displayLocale);
+
+        if (isset($localeBundle['zoneStrings']) && null !== $localeBundle['zoneStrings']) {
+            $data = array(
+                'Version' => $localeBundle['Version'],
+                'Names' => $this->generateTimezoneNames($localeBundle),
+            );
+
+            $this->timezones = array_merge($this->timezones, array_keys($data['Names']));
+
+            return $data;
+        }
+
+        return;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForRoot(BundleReaderInterface $reader, $tempDir)
+    {
+        $rootBundle = $reader->read($tempDir, 'root');
+
+        $names = $this->generateTimezoneNames($rootBundle);
+
+        foreach ($this->timezones as $timezone) {
+            if (!isset($names[$timezone])) {
+                $names[$timezone] = $this->getFallbackName($timezone);
+            }
+        }
+
+        ksort($names);
+
+        return array(
+            'Version' => $rootBundle['Version'],
+            'Names' => $names,
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function generateDataForMeta(BundleReaderInterface $reader, $tempDir)
+    {
+        $rootBundle = $reader->read($tempDir, 'root');
+
+        $this->timezones = array_unique($this->timezones);
+
+        sort($this->timezones);
+
+        $data = array(
+            'Version' => $rootBundle['Version'],
+            'Timezones' => $this->timezones,
+        );
+
+        return $data;
+    }
+
+    /**
+     * @param ArrayAccessibleResourceBundle $rootBundle
+     *
+     * @return array
+     */
+    private function generateTimezoneNames(ArrayAccessibleResourceBundle $rootBundle)
+    {
+        $timezoneNames = array();
+        foreach ($rootBundle['zoneStrings'] as $key => $zone) {
+            if (strpos($key, ':') !== false && substr($key, 5) != 'meta:') {
+                $identifier = str_replace(':', '/', $key);
+
+                $zone = iterator_to_array($zone);
+
+                if (isset($zone['ec'])) {
+                    $timezoneNames[$identifier] = $zone['ec'];
+                }
+            }
+        }
+
+        return $timezoneNames;
+    }
+
+    /**
+     * Converts a timezone identifier to an English string.
+     *
+     * @param string $timezone
+     *
+     * @return string
+     */
+    private function getFallbackName($timezone)
+    {
+        $parts = explode('/', $timezone);
+        if (count($parts) > 2) {
+            $name = $parts[2].', '.$parts[1];
+        } elseif (count($parts) > 1) {
+            $name = $parts[1];
+        } else {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method getFallbackName uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://www.codacy.com/app/git_9/symfony/file/2332020086/issues/source?bid=3207821&fileBranchId=3207896#l157)